commons/com.b2international.index/src/com/b2international/index/revision/RevisionConflictProcessor.java

```diff
@@ -91,12 +105,104 @@ default Conflict convertConflict(Conflict conflict) {
 	class Default implements RevisionConflictProcessor {
 		
 		@Override
-		public RevisionPropertyDiff handleChangedInSourceAndTarget(String revisionId, RevisionPropertyDiff sourceChange, RevisionPropertyDiff targetChange) {
-			// apply source change if the new value is the same, otherwise report conflict
-			if (Objects.equals(sourceChange.getNewValue(), targetChange.getNewValue())) {
+		public RevisionPropertyDiff handleChangedInSourceAndTarget(String revisionId, DocumentMapping mapping, RevisionPropertyDiff sourceChange, RevisionPropertyDiff targetChange, ObjectMapper mapper) {
+			String property = sourceChange.getProperty();
+			if (mapping.isObject(property)) {
+				// in case of Object properties, allow changes touching different properties to be merged without reporting conflict 
+				return handleObjectConflict(mapping, sourceChange, targetChange, mapper);
+			} else if (mapping.isCollection(property)) {
+				// in case of Collection/Array properties, allow subsets to be merged together 
+				// eg. [1,2] -> [1,2,3] vs. [1,2] -> [1,2,4] should not produce conflicts, but merge the array into [1,2,3,4]
+				return handleCollectionConflict(mapping, sourceChange, targetChange, mapper, mapping.isSet(property));
+			} else if (Objects.equals(sourceChange.getNewValue(), targetChange.getNewValue())) {
```

Comment:
No, it's not. The `getNewValue()` method returns the String representation of the value. In case of simple types like primitives and Strings this works fine, but when it comes to arrays/objects it needs extra care if the array/object JSON node list changes, but semantically the same (eg. order change). We don't want to report a conflict when an array changes from [1, 2] -> [2, 1] or when an object property list order changes, but holds the same values.